patches/server/0715-Add-API-for-pausing-awareness-and-AI.patch

```diff
@@ -0,0 +1,271 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: froobynooby <froobynooby@froobworld.com>
+Date: Sun, 20 Jun 2021 14:28:29 +0930
+Subject: [PATCH] Add API for pausing awareness and AI
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/Mob.java b/src/main/java/net/minecraft/world/entity/Mob.java
+index c4c5c35e37b793f3b74349ff03c0829f4913b91c..9c9e61c0febf10e9945cc02d70436ce381e2f3f5 100644
+--- a/src/main/java/net/minecraft/world/entity/Mob.java
++++ b/src/main/java/net/minecraft/world/entity/Mob.java
+@@ -125,6 +125,18 @@ public abstract class Mob extends LivingEntity {
+     private float restrictRadius;
+ 
+     public boolean aware = true; // CraftBukkit
++    // Paper start
++    public java.util.Set<String> aiPausedPlugins = new java.util.HashSet<>();
++    public java.util.Set<String> awarenessPausedPlugins = new java.util.HashSet<>();
++
++    public boolean pausedOrNoAI() {
++        return this.isNoAi() || !this.aiPausedPlugins.isEmpty();
++    }
```

Comment:
Instead of adding this method, and then changing all the other calls to isNoAI, just change what the isNoAI method does